### PR TITLE
Changes to allow applications to set OIDC scopes

### PR DESCRIPTION
**Pull request by kreedula**

### CODE REVIEW

**Comment by shahzaibj** [comment]
So looks like this is targeted for the current broker release?

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java [hunk 1]
```diff
@@ -469,8 +469,12 @@ protected boolean idTokenIsNull(@NonNull final ICacheRecord cacheRecord,
         return null == idTokenRecord;
     }
 
-    protected Set<String> addDefaultScopes(@NonNull final TokenCommandParameters commandParameters) {
+    protected Set<String> addDefaultAndOidcScopes(@NonNull final TokenCommandParameters commandParameters) {
         final Set<String> requestScopes = commandParameters.getScopes();
+
+        if (commandParameters.getOidcScopes() != null && !commandParameters.getOidcScopes().isEmpty()) {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java:475]: TokenCommandParameters::getScopes() can pretty explicitly return null.  Should we handle that?

File: common/src/main/java/com/microsoft/identity/common/internal/commands/parameters/TokenCommandParameters.java [hunk 2]
```diff
@@ -61,6 +61,9 @@
     @Expose()
     private boolean forceRefresh;
 
+    @Expose()
+    private Set<String> oidcScopes;
+
     public Set<String> getScopes() {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/commands/parameters/TokenCommandParameters.java:67]: Maybe add @Nullable to getScopes(), or is that not an actual possibility?

**Comment by iambmelt** [comment]
Can we add a description to PR covering what's being added, how/why?

**Comment by kreedula** [comment]
Closing this as this is no longer required for Teams

**Pull request closed without merging**